src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -2416,4 +2421,20 @@ private EndpointState removeEndpointState(InetAddressAndPort endpoint)
         return removedState;
     }
 
+    private void tryTaskLock()
```

Comment:
`void try` sounds weird. 
You try to do something, but what happens when the try is unsuccessful?

Since we get an assertion failure if the try fails, I'd just call the function `taskLock`; Assertion failure is not a part of logic.
Alternatively, we might use sth along `lockOrThrow` etc, but this would indicate that throwing is a part of logic that should be handled (and it is not)